bears/general/FilenameBear.py

```diff
@@ -37,8 +37,14 @@ def run(self, filename, file,
         head, tail = os.path.split(filename)
         filename_without_extension, extension = os.path.splitext(tail)
         try:
+            if extension == '.java':
+                file_naming_convention = 'pascal'
+
             new_name = self._naming_convention[file_naming_convention](
                 filename_without_extension)
+
+            if extension == '.js':
+                new_name = new_name.replace('_', '-')
```

Comment:
the JS style should have its own proper name defined in the dict.

Create a separate issue & PR, as required, as this would be a feature.